testing.py

```diff
@@ -422,6 +424,37 @@ def _complaint(self, license_pool, type, source, detail, resolved=None):
         )
         return complaint
 
+    def _credential(self, data_source_name=DataSource.GUTENBERG,
+                    type=None, patron=None):
+        data_source = DataSource.lookup(self._db, data_source_name)
+        type = type or self._str
+        patron = patron or self._patron()
+        credential, is_new = Credential.persistent_token_create(
+            self._db, data_source, type, patron
+        )
+        return credential
+    
+    def _delegated_patron_identifier(
```

Comment:
In an earlier version of this branch, I had DRMDeviceIdentifier associated with DelegatedPatronIdentifier, and I needed this helper method to get the tests to work. I moved away from that design later, but this helper method seems generally useful, so I kept it.